### PR TITLE
rename the --develop option to --debug

### PR DESCRIPTION
**Pull request by devoncarew**
- rename the `--develop` option to `--debug`
- rename `BuildVariant` to `BuildMode`
- add `--debug` and `--deploy` flags to `flutter run`

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -60,6 +60,22 @@ abstract class FlutterCommand extends Command {
     _usesPubOption = true;
   }
 
+  void addBuildModeFlags() {
+    argParser.addFlag('debug',
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: having these be flags means it's easy to end up specifying mutually-exclusive flags.
Maybe `--build-mode=debug` or some such?
But that's non-ergonomic.
If we stick with flags, we should be very elaborate in our handling of the situation where mutually exclusive flags are given at the same time.

**Comment by Hixie** [comment]
LGTM.
I'd go with "release" rather than "deploy". Searching google for "deploy mode" seems to suggest it's used primarily to decide _where_ to deploy, not _what_ to deploy, in existing products.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: `--build-mode=debug` would be clearer but more to type. I'll add some checks in the `getBuildMode()` call below to handle mutually exclusive flags.

**Comment by devoncarew** [comment]
I'll need to change the chromium bots script to rename the bucket we're uploading to, and then coordinate that w/ a PR into this repo. I'd like to handle that in a separate PR to make things a bit easier.

**Pull request merged**